src/plugins/expressions/common/expression_functions/specs/kibana.ts

```diff
@@ -44,15 +44,15 @@ export const kibana: ExpressionFunctionKibana = {
 
   args: {},
 
-  fn(input, _, { search = {} }) {
+  fn(input, _, { getSearchContext }) {
     const output: ExpressionValueSearchContext = {
       // TODO: This spread is left here for legacy reasons, possibly Lens uses it.
       // TODO: But it shouldn't be need.
       ...input,
       type: 'kibana_context',
-      query: [...toArray(search.query), ...toArray((input || {}).query)],
-      filters: [...(search.filters || []), ...((input || {}).filters || [])],
-      timeRange: search.timeRange || (input ? input.timeRange : undefined),
+      query: [...toArray((getSearchContext() as any).query), ...toArray((input || {}).query)],
+      filters: [...((getSearchContext() as any).filters || []), ...((input || {}).filters || [])],
+      timeRange: (getSearchContext() as any).timeRange || (input ? input.timeRange : undefined),
```

Comment:
Feels like we shouldn't need to cast to `any` here -- presumably `getSearchContext()` is getting its type from `ExecutionContextSearch`, in which case query, filters, timeRange are all undefined so you should still be able to use them right?